File: Python/sieve.py
def sieve(n):
  prime = [True for i in range(n+5)]
  prime[0]=False
  prime[1]=False
  pnt = 2
  while(pnt<=n**0.5):
    if prime[pnt]==True:
      for j in range(pnt*pnt,n+5,pnt):
        prime[j]=False
    pnt+=1
  
  for i in range(2,n):
    if prime[i]==True:
      print(i)

File: c++/sieve.cpp
//Sieve of Erastothenes

#include<iostream>
#include<conio.h>
#include<math.h>

using namespace std;

int main()
{
 //clrscr();
 int n,f=0;
 cout<<"Enter the range ";
 cin>>n;
 int p[100],m=0,r=sqrt(n),q=2;

 for(int i=1;i<=n;++i)
 { f=0;
  for(int j=q;j<=r;++j)
  { if(j==i) {f=1;}
    else if(i%j==0){f=0; break;}
    else f=1;
  }
  if(f==1) cout<<i<< " "; //{p[m]=i; ++m;}
 }

  return 0;
  }
